main/src/ca/uwaterloo/flix/language/phase/unification/set/SetUnification.scala

```diff
@@ -17,29 +17,284 @@
 package ca.uwaterloo.flix.language.phase.unification.set
 
 import ca.uwaterloo.flix.language.phase.unification.set.SetFormula.*
+import ca.uwaterloo.flix.util.Result
+
+import scala.collection.mutable
 
 object SetUnification {
 
   /**
     * The static parameters of set unification.
     *
-    * @param sveRecSizeThreshold if positive, [[sve]] will give up on formulas beyond this size.
+    * @param sizeThreshold if positive, [[solveWithInfo]] will give up before SVE if there are more
+    *                      equations than this
+    * @param permutationLimit if positive, the maximum number of permutations that
+    *                         [[svePermuations]] will try
+    * @param sveRecSizeThreshold if positive, [[sve]] will give up on formulas beyond this size
+    * @param svePermutationExitSize [[svePermuations]] will stop searching early if it finds a\
+    *                               substitution smaller than this
+    * @param debugging the amount of debugging information printed
+    */
+  final case class Options(
+                            sizeThreshold: Int = 1800,
+                            permutationLimit: Int = 10,
+                            sveRecSizeThreshold: Int = 1800 * 2,
+                            svePermutationExitSize: Int = 0,
+                            debugging: Options.Debugging = Options.Debugging.Nothing
+                          )
+
+  object Options {
+    sealed trait Debugging
+
+    object Debugging {
+      case object Nothing extends Debugging
+
+      case object All extends Debugging
+    }
+  }
+
+  /**
+    * Represents the running state of the solver.
+    *   - `eqs`: the remaining equations to solve
+    *   - `errEqs`: the remaining equations that have been found unsolvable
+    *   - `subst`: the current substitution, which has already been applied to `eqs`
+    *   - `lastProgressPhase`: the name of the last phase that was run and made progress
+    *   - `currentPhase`: the number of the last phase that was run
+    */
+  private class State(var eqs: List[Equation]) {
```

Comment:
Why mix the styles? Initially I could not find eqs because you declared it in the class signature.

But you declare errEqs in the class body.

Reply:
I would make it consistent.